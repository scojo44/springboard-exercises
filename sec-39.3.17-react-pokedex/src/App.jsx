import React from 'react'
import Pokegame from './Pokegame';
import './App.css'

// const App = () => <Pokegame />;
function App() {
  return <Pokegame />;
}

export default App;
