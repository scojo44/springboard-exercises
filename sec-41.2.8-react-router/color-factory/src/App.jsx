import React from 'react'
import {Link} from 'react-router-dom'
import RouteList from './RouteList'

const App = () => (
  <>
    <h1><Link to="/colors">Color Factory</Link></h1>
    <RouteList />
  </>
);

export default App
