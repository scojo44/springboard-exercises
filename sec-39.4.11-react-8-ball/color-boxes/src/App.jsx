import React from 'react'
import ColorPallet from './ColorPallet';
import './App.css'

function App() {
  return (
    <>
      <h1>Color Boxes</h1>
      <ColorPallet />
    </>
  );
}

export default App
