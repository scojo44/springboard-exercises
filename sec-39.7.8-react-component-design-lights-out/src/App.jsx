import React from "react";
import Board from "./Board";
import "./App.css";

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
      <div className="App">
        <h1>Lights Out!</h1>
        <Board rows={10} cols={10} />
      </div>
  );
}

export default App;
