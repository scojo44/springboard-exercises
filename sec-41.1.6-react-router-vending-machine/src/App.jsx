import React from 'react'
import VendingMachine from './VendingMachine'
import './App.css'

function App() {
  return (
    <VendingMachine />
  )
}

export default App
