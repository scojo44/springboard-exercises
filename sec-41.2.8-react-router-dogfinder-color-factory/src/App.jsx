import React from 'react'
import { Route, Routes, Navigate } from 'react-router-dom'
import {dogs} from './dogs.json'
import NavBar from './NavBar'
import DogList from './DogList'
import DogDetails from './DogDetails'
import './App.css'

function App() {
  return (
    <>
      <NavBar dogNames={dogs.map(d => d.name)} />
      <Routes>
        <Route exact path="/dogs" element={<DogList dogs={dogs} />}/>
        <Route path="/dogs/:id" element={<DogDetails dogs={dogs} />}/>
        <Route path="*" element={<Navigate to="/dogs" />}/>
      </Routes>
    </>
  )
}

export default App
