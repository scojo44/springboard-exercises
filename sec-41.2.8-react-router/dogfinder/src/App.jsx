import React from 'react'
import {dogs} from './dogs.json'
import RouteList from './RouteList'
import NavBar from './NavBar'
import './App.css'

function App() {
  return (
    <>
      <NavBar dogNames={dogs.map(d => d.name)} />
      <RouteList />
    </>
  )
}

export default App
