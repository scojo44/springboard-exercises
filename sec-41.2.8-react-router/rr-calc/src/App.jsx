import React from 'react'
import {BrowserRouter, Routes, Route} from 'react-router-dom'
import Calc from './Calc'
import './App.css'

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/:op/:a/:b" element={<Calc />} />
      </Routes>
    </BrowserRouter>
  )
}

export default App
