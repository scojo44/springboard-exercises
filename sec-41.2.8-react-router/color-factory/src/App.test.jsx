import {render} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from './App'

describe('App Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Integration Tests', () => {
  it('Redirects to the color list if color not found', () => {
    const {getByText} = render(
      <MemoryRouter initialEntries={['/colors/nope']}>
        <App />
      </MemoryRouter>
    );
    expect(getByText('Add a new color')).toBeInTheDocument();
  });
});
