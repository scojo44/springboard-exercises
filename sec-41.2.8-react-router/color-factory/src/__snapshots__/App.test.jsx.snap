// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Color Factory
  </h1>
  <div
    class="ColorList"
  >
    <p>
      <a
        href="/colors/new"
      >
        Add a new color
      </a>
    </p>
    <menu />
  </div>
  <p>
    <a
      href="/colors"
    >
      Home
    </a>
  </p>
</DocumentFragment>
`;
